keras_nlp/models/gpt_neox/gpt_neox_backbone.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.experimental import dtensor
+from tensorflow.experimental.dtensor import Layout
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.backbone import Backbone
+from keras_nlp.models.gpt_neox.gpt_neox_decoder import GPTNeoXDecoder
+
+
+def _gpt_neox_kernel_initializer(stddev=0.02):
+    return keras.initializers.RandomNormal(stddev=stddev)
+
+
+@keras_nlp_export("keras_nlp.models.GPTNeoXBackbone")
+class GPTNeoXBackbone(Backbone):
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        hidden_dim,
+        intermediate_dim,
+        dropout=0.,
+        rotary_pct=0.25,
+        rotary_emb_base=10000,
+        layer_norm_epsilon=1e-5,
+        max_sequence_length=512,
```

Comment:
this appears unused, did you mean to pass this down to the decoder?

Reply:
Yes I meant to pass it down, I think I missed it.